### PR TITLE
Take a URL from config value to fetch institution list for Registry 

### LINKED ISSUE #635: Check uniqueness of institution IDs from config

### LINKED ISSUE #648: Enable admin to pass a URL for fetching institution list for registry
Currently, for a list of institution available for user to register to (used by registry UI when user register their namespace for listing options for institutions), we get it from config value `Registry.Institutions`, a yaml array of institutions. 

We want to make it more flexible by allowing admin to feed a URL with a remote endpoint providing such list.

### PR DESCRIPTION
**Pull request by haoming29**
Fixes #635 and Closes #648 

To test locally, you may use the topology URL which gives valid institution list for OSDF: https://topology.opensciencegrid.org/institution_ids

### CODE REVIEW

File: docs/parameters.yaml [hunk 1]
```diff
@@ -600,10 +599,48 @@ description: >-
...
   ```
 
+  Note that this value will take precedence over Registry.InstitutionsUrl if both are set
 type: object
 default: none
 components: ["nsregistry"]
 ---
+name: Registry.InstitutionsUrl
+description: >-
+  A url to get a list of available institutions for users to regiter their namespaces to.
```
Comment (jhiemstrawisc) [inline docs/parameters.yaml:609]: `regiter --> register`

File: registry/registry_ui_test.go [hunk 2]
```diff
@@ -267,3 +279,311 @@ func TestPopulateRegistrationFields(t *testing.T) {
...
+		unique := checkUniqueInstitutions([]Institution{{ID: "1"}, {ID: "2"}})
+		assert.True(t, unique)
+	})
+
+	t.Run("duplicated-gives-false", func(t *testing.T) {
+		unique := checkUniqueInstitutions([]Institution{{ID: "1"}, {ID: "1"}})
+		assert.False(t, unique)
+	})
+
+	t.Run("large-entries", func(t *testing.T) {
```
Comment (jhiemstrawisc) [inline registry/registry_ui_test.go:439]: What does this test check that the previous doesn't?

File: registry/registry_ui.go [hunk 3]
```diff
@@ -575,3 +682,67 @@ func RegisterRegistryWebAPI(router *gin.RouterGroup) error {
...
+			} else {
+				log.Info("Institution TTL cache is nil, stop clean up process.")
+			}
+			return nil
+		})
+
+		// Try to populate the cache at the server start. If error occured, it's non-blocking
+		cachedInsts, intErr, _ := getCachedInstitutions()
+		if intErr != nil {
+			log.Warning("Failed to populate institution cache. It's non-blocking. Error: ", intErr)
```
Comment (jhiemstrawisc) [inline registry/registry_ui.go:734]: I think it's redundant to point out this is non-blocking if it's already in a `log.Warning`.

**Review by jhiemstrawisc — APPROVED** [review]
There's one typo that should be cleaned up, and two small comments that you can choose to ignore if you want. Otherwise LGTM. Once the typo is fixed, I'll approve and merge.

File: registry/registry_ui_test.go (hunk 2 above)
Comment (haoming29) [reply registry/registry_ui_test.go:439]: The idea was to add some relatively "large" number of entries to make sure we are not just reading the first couple of entries to check for duplicates

**Pull request merged**